### PR TITLE
[task-6] As an User, I can land on Screen for Login

### PR DESCRIPTION
**Pull request by noor-ahmed**
## What happened
- [x] basic component for 'Log in"
- [x] Login Screen
- [x] Test coverage
 
## Insight
[Feature](https://github.com/noor-ahmed/nimble-survey/projects/1#card-43838918)

## Proof Of Work
Screen Shot 2020-10-14 at 7 44 44 PM

**Responsiveness** 
Screen Shot 2020-10-14 at 7 45 08 PM

 - Test should passed

### CODE REVIEW

File: src/routes/index.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+const RouteList = () => {
+  return (
+    <Switch>
+      <Route exact path="/" component={Login}></Route>
+    </Switch>
+  );
+};
+
+export default RouteList;
```
Comment (Lahphim) [inline src/routes/index.jsx]: @noor-ahmed Can you add an extra line here? 😅

File: src/screens/Login/form.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+function Form() {
+  return (
+    <div className="form">
+      <div className="form-group__container">
+        <label id="email" htmlFor="email" className="form-group__label">
```
Comment (Lahphim) [inline src/screens/Login/form.jsx]: @noor-ahmed Is this the way to link `label` and `input` like this below?
https://www.w3schools.com/tags/att_input_type_text.asp

`for` attribute from `label` will link to `id` for an `input` tag

File: src/screens/Login/form.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+function Form() {
+  return (
+    <div className="form">
```
Comment (Lahphim) [inline src/screens/Login/form.jsx]: @noor-ahmed Should we use `form` here instead of `div`?  🤔
I think it'd be nice if we use `form`'s behavior.

File: src/screens/Login/form.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+      <div className="form-group">
+        <label id="password" htmlFor="password" className="form-group__label">
+          Password
+        </label>
+        <input type="password" name="password" className="form-group__input" />
+      </div>
+
+      <div className="footer">
+        <button type="button" className="btn button--white">
```
Comment (Lahphim) [inline src/screens/Login/form.jsx]: @noor-ahmed Once we change it to form, I think we should change the type of this button to `submit` 🙌

File: src/routes/index.jsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import { Switch, Route } from 'react-router-dom';
+
+import { Login } from '../screens/Login';
+
+const RouteList = () => {
+  return (
+    <Switch>
+      <Route exact path="/" component={Login}></Route>
```
Comment (Lahphim) [inline src/routes/index.jsx:10]: @noor-ahmed Between `{Login}` and `{ Login }`, which one should we go with it? 😅

File: src/stylesheets/_variables.scss [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  primary: hsl(0, 0%, 100%)
+);
+
+$brand-gray: (
+  transparent: hsla(0, 0%, 15%, 0.6)
+);
+
+// Font Size
+$font-sizes: (
+  sm: 0.8125rem,
```
Comment (Lahphim) [inline src/stylesheets/_variables.scss]: @noor-ahmed Should we use `rem()` function here also?
If we cannot use it here, we should add some comment after this line to tell how the size of the pixel for this `rem`. 👍

File: src/stylesheets/base/_buttons.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+.button {
+  @extend .btn;
+
+  &--white {
```
Comment (Lahphim) [inline src/stylesheets/base/_buttons.scss:4]: @noor-ahmed All of the properties in here look not related to its class name 🤔
Should we move some properties which are not about color to another new class?
And we leave just only properties about color here 😅

File: src/stylesheets/components/_form-group.scss [hunk 8]
```diff
@@ -0,0 +1,61 @@
+.form-group {
```
Comment (Lahphim) [inline src/stylesheets/components/_form-group.scss:1]: @noor-ahmed I think we have this class in bootstrap already.
Is this about overriding? 🤔

File: src/stylesheets/vendors/bootstrap.scss [hunk 9]
```diff
@@ -0,0 +1,13 @@
+@import '~bootstrap/scss/functions';
```
Comment (Lahphim) [inline src/stylesheets/vendors/bootstrap.scss:1]: @noor-ahmed Should we add all in the bootstrap here and comment to those components which are we don't use it?
Once we want to use that we just un-comment that line. 🤔

File: src/tests/routes/index.test.jsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    const history = createMemoryHistory();
+    render(
+      <Router history={ history }>
+        <Route exact path="/" component={Login}></Route>
+      </Router>
+    );
+
+    expect(history.location.pathname).toBe('/');
+  });
+});
```
Comment (Lahphim) [inline src/tests/routes/index.test.jsx]: @noor-ahmed Please add an extra line here 🧩

File: src/routes/index.jsx (hunk 5 above)
Comment (noor-ahmed) [reply src/routes/index.jsx:10]: `{ Login }` is 👌

File: src/stylesheets/components/_form-group.scss (hunk 8 above)
Comment (noor-ahmed) [reply src/stylesheets/components/_form-group.scss:1]: Yes.. for fulfilling the requirements we had to do this. Should I write another class?

File: src/screens/Login/form.jsx (hunk 3 above)
Comment (noor-ahmed) [reply src/screens/Login/form.jsx]: updated 🙌

File: src/routes/index.jsx (hunk 1 above)
Comment (noor-ahmed) [reply src/routes/index.jsx]: Opps.. added 😃

File: src/stylesheets/base/_buttons.scss (hunk 7 above)
Comment (noor-ahmed) [reply src/stylesheets/base/_buttons.scss:4]: Created this file https://github.com/noor-ahmed/nimble-survey/pull/6/commits/51474e7e7ee9b579de754f72866c2701dae25f91#diff-05b7fe7d0327e6cabe8681914cb1c2511cbb9c3c90ec14181577d76bad055065

File: src/stylesheets/components/_logo-base.scss [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  &__img {
+    align-self: center;
+    margin: rem(0 24px 24px);
+  }
+
+  &__label {
+    align-self: stretch;
+    color: map-get($brand-white, 'primary');
+    font-size: map-get($font-sizes, 'lg');
+    letter-spacing: -0.41px;
```
Comment (rafayet-monon) [inline src/stylesheets/components/_logo-base.scss]: I think we can get rid of the negative letter spacing.

File: src/stylesheets/components/_submit_button.scss [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+  @extend .btn;
+
+  align-items: center;
+  border-radius: rem(10px);
+
+  font-size: map-get($font-sizes, 'lg');
+  font-weight: map-get($font-weight, 'bolder');
+
+  height: rem(55px);
+  left: calc(50% - 343px/2 - 0.5px);
```
Comment (rafayet-monon) [inline src/stylesheets/components/_submit_button.scss]: Should we use rem in these places as well?

File: src/stylesheets/components/_form-group.scss [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    position: static;
+    left: 0%;
+    right: 0%;
+    top: 0%;
+    bottom: 76.19%;
+
+    font-style: normal;
+    font-weight: 600;
+    font-size: 15px;
+    line-height: 20px;
```
Comment (rafayet-monon) [inline src/stylesheets/components/_form-group.scss]: `rem()` in these two lines.

File: src/stylesheets/components/_form-group.scss [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+  &__label {
+    position: static;
+    left: 0%;
+    right: 0%;
+    top: 0%;
+    bottom: 76.19%;
+
+    font-style: normal;
+    font-weight: 600;
```
Comment (rafayet-monon) [inline src/stylesheets/components/_form-group.scss]: Can we access this font-weights from `map-get`?

File: src/stylesheets/components/_form-group.scss [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+  flex-direction: column;
+  width: rem(320px);
+  height: rem(90px);
+  margin-bottom: rem(20px);
+
+  &__label {
+    position: static;
+    left: 0%;
+    right: 0%;
+    top: 0%;
```
Comment (rafayet-monon) [inline src/stylesheets/components/_form-group.scss]: can we get rid of the 0%?

**Review by rafayet-monon — APPROVED** [review]

**Review by Lahphim — APPROVED** [review]
